openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -315,6 +318,10 @@ def _handle_successful_authentication_and_login(user, request):
         request.session.set_expiry(604800 * 4)
         log.debug("Setting user session expiry to 4 weeks")
 
+        if settings.ENFORCE_SESSION_EMAIL_MATCH:
```

Comment:
I wonder if we can drop all of these conditions, and just have the condition protect the flushing of the session? It would be a little riskier, because we'd still be messing with code that could break. But, it would allow observability into what would be enforced before we flip the switch to actually start killing sessions. Something to consider. This would apply to any of these conditionals that come before the final enforcement (and the cookie updates).

Reply:
I have moved the cookies update logic to `process_response` of middleware and moved it behind a variable `session_and_user_email_mismatch` that is marked true in the `register_email_change` function of the middleware PLUS the ENFORCE_SESSION_EMAIL_MATCH flag check. 
Now we have the `ENFORCE_SESSION_EMAIL_MATCH` in the `process_request` and `process_response` function of middleware. Could you please provide your opinion on this?

Reply:
Yes. I missed that there were two things that needed to be protected, the flushing and the updating of the cookies. Looks good.